src/algorithms/bnb.rs

```diff
@@ -1,13 +1,22 @@
 use rand::{rngs::ThreadRng, thread_rng, Rng};
 
 use crate::{
-    types::{
-        CoinSelectionOpt, MatchParameters, OutputGroup, SelectionError, SelectionOutput,
-        WasteMetric,
-    },
+    types::{CoinSelectionOpt, OutputGroup, SelectionError, SelectionOutput, WasteMetric},
     utils::{calculate_fee, calculate_waste, effective_value},
 };
 
+/// Struct for three arguments : target_for_match, match_range and target_feerate
+///
+/// Wrapped in a struct or else input for fn bnb takes too many arguments - 9/7
+/// Leading to usage of stack instead of registers - https://users.rust-lang.org/t/avoiding-too-many-arguments-passing-to-a-function/103581
```

Comment:
This was added by me, but sounds a bit unprofessional, plus a link in the comments. Should it be removed?

Reply:
I noticed it, can you come up with good comment?
It's a private struct we can just remove the comments.
It's not that complicated though.

Reply:
I will come up with a better comment

Reply:
/// Struct MatchParameters encapsulates target_for_match, match_range, and target_feerate.
/// Reduces the number of arguments in fn bnb, optimizing register usage and minimizing stack allocation.
/// Fits within 1 XMM register and 1 GPR for improved performance.
/// Link for reference : https://users.rust-lang.org/t/avoiding-too-many-arguments-passing-to-a-function/103581